### ISSUE TITLE
The model predicts only spaces?

### ISSUE THREAD

**Issue by Nitinkmr** [sagar448/Keras-Recurrent-Neural-Network-Python#1]
The array randomStart has all spaces in it at the end even if i try with some other dataset and different epochs?

**Comment by phyden** [comment]
Yes, that's the output for me too. I think, that the problem is, that letters should not be converted to floats but be categorical binary vectors. You are using categories for y, so why not for X too?

I changed that, and trained it for 100 epochs with 128 LSTM units instead of 256 and get at least some text output - which somehow resembles some words partially.. 
e.g. "sthe tant oo hean bate hear thould bate she t sos paneis ithave share,
of andris path"

`#This is done for normalization`
`X = np_utils.to_categorical(X,numberOfUniqueChars)`

and later on in the for loop after np.reshape again
`x = np_utils.to_categorical(x,numberOfUniqueChars)
`

I'm not an expert on the field, but I guess that with the limited training data and the way how the random text is generated this is already quite a decent result, considering that a LSTM large enough was easily memorizing the whole text as such (I got that result with 256 LSTM units in 100 epochs). However, some fine tuning might still get you slightly better results.

**Comment by sagar448** [comment]
Actually, let me make it a bit clearer. I am to blame yes as I did not put this in my tutorial but, the data is not enough. Consider this, copy the text that i've provided and paste it about 20-30 times. Now, obviously this is "hardcoded" training, but you'd start seeing some results. Why? Well I believe the answer is, the algorithm doesn't like it if the data size is anything below 20kb. Hope this helps.. 

Im sure Phyden's corrections may have helped, but they are not necessary.
